source/material_model/rheology/elasticity.cc

```diff
@@ -119,11 +119,33 @@ namespace aspect
       Elasticity<dim>::parse_parameters (ParameterHandler &prm)
       {
         // Get the number of fields for composition-dependent material properties
-        const unsigned int n_fields = this->n_compositional_fields() + 1;
-
-        elastic_shear_moduli = Utilities::possibly_extend_from_1_to_N (Utilities::string_to_double(Utilities::split_string_list(prm.get("Elastic shear moduli"))),
-                                                                       n_fields,
-                                                                       "Elastic shear moduli");
+        const bool has_background_field = true;
+
+        stress_field_indices = this->introspection().get_field_type_indices().stress;
+        compositional_field_indices = this->introspection().get_field_type_indices().composition;
+
+        AssertThrow (stress_field_indices.size() == (dim == 2 ? 3 : 5),
+                     ExcMessage("The Compositional fields subsection must contain the parameter "
+                                "Types of fields. Additionally, there must be three or five fields "
+                                "corresponding to stress (for two- or three-dimensional problems respectively)."));
+
+        const std::vector<std::string> list_of_field_names = this->introspection().get_composition_names();
+
+        std::vector<std::string> list_of_composition_names(compositional_field_indices.size());
+        std::transform(compositional_field_indices.begin(),
```

Comment:
This function will probably be used a lot, maybe it should go into Utilities?

Reply:
Thanks @anne-glerum! I agree. The changes that @wolfgang and @gassmoeller proposed at our last meeting will result in a few new utilities functions. I'll try to shift as many multi-use functions out of the material models as possible.

Reply:
Good to know! Too bad I couldn't make it to the meeting.